components/sidebar/sidebar.jsx

```diff
@@ -566,6 +569,24 @@ export default class Sidebar extends React.PureComponent {
                     id='sidebarChannelContainer'
                     className='nav-pills__container'
                 >
+                    {pluginApps && pluginApps.length > 0 &&
```

Comment:
The idea of this block is to have the concept of "Application", that is embeded inside mattermost and show that applications in the LHS.

Reply:
Maybe, but the problem is, when you write an application deeply integrated, you may want to show things in the channel screen when you click in the screen header buttons, and things in the team screen when you click the "Apps" menu button, but it can be different for each plugin.

I prefer to have separated building blocks (exposed in the API) and some combination of this building blocks. for example, have `registerAppMenuItem`, `registerChannelScreen`, `registerChannelHeaderItem`, etcetera. And then have a `registerTeamApp` which register everything needed for a Team App, like the AppMenuItem, or the TeamScreen, or whatever. And maybe a `registerApp` which allow you to set the most common screens (team and channel), and add the needed menu items where needed.

Reply:
0/5 here but does it ever make sense to register a button in the LHS that doesn't go to a channel screen? I feel like having LHS buttons that don't perform a changing channel (or changing to some other screen in the center channel) would be potentially problematic from a technical and UX point of view

Reply:
I think so. For example, if you have something which is "global" to the team. In the Kanba board example, you have the "Team kanban" linked in the LHS menu inside the Application section, and is scoped to team level, not channel level.

The channel header is another thing. If you have something there, it is clearly related to the current channel, and, in case it show a screen there, it will be channel scoped screen (IMHO).

Reply:
Yes, I my opinion makes sense in some cases. You can want to make an action that is global to the system and maybe is triggered using a modal, not the center channel. And maybe is just a like to somewhere. An example could be something like an integration with Calendar that shows there the current public live meetings in zoom for mattermost stuff. And maybe private ones if you have a @mattermost.com email.

So, you can see there a link to the zoom screen from 10 minutes before the meeting to the end of the meeting.

Reply:
Ok, no objection here 👍